### ISSUE TITLE
[P0] DeepSpeedCPUAdam not compiled — A6000 falls back to torch AdamW, 4x slower optimizer step

### ISSUE THREAD

**Issue by dylanyunlon** [dylanyunlon/Neuron_SP#84]
## Symptom
On ags1 (CUDA 13.0, Driver 580), `deepspeed.ops.adam.cpu_adam` fails to build because `csrc/adam/cpu_adam_impl.cpp` uses AVX2 intrinsics that are available but the build system cannot find the correct include paths for EPYC 9354.

Fallback: `torch.optim.AdamW` with `fused=True` on GPU. This works but is ~4x slower than CPU offloaded Adam for the A6000 workers (which have abundant CPU bandwidth via PCIe Gen4).

## API surface
```python
# Current fallback path in desloc_engine.py:
try:
    from deepspeed.ops.adam import DeepSpeedCPUAdam
except ImportError:
    # Falls back to torch.optim.AdamW
```

## Fix approach
1. Fix the build: `op_builder/cpu_adam.py` needs CUDA 13.0 compatibility flags
2. Alternatively: use `torch.optim.AdamW(fused=True)` on GPU but offload states to CPU manually via ZeRO-3 offload

## Labels
Module: core/optimizer, Priority: P0
